ksqldb-engine/src/test/java/io/confluent/ksql/function/UdfLoaderTest.java

```diff
@@ -1000,7 +1000,7 @@ public void shouldThrowIfMissingAggregateTypeSchema() throws Exception {
     assertThat(e.getMessage(), containsString("Must specify 'aggregateSchema' for STRUCT parameter in @UdafFactory."));
   }
 
-  @Test
+  //@Test
```

Comment:
Ok, another thing to discuss... functions either need an annotated type or compute aggregate/return types.  

I'll think about these more next week.  Shout if you have an options about how to sort this out.

Reply:
😅  😆  😈  I know I promised I'd drop it, but if we went back to having a wrapper on the three methods you could check if calling that returns the default implementation or not

Reply:
This test / behavior is changing since one can either have types given with an annotation or via the new methods.  

If we had a wrapper, we'd still be in the same situation between annotations and the wrapper.  🤷 

I'll see what I come up when I look again.

Reply:
Ok, I've sorted this out by adding a little to the error to indicate choice in implementation.